src/main/java/me/itzg/helpers/modrinth/ModrinthPackInstaller.java

```diff
@@ -0,0 +1,221 @@
+package me.itzg.helpers.modrinth;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.StandardCopyOption;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipInputStream;
+
+import lombok.extern.slf4j.Slf4j;
+import me.itzg.helpers.errors.GenericException;
+import me.itzg.helpers.errors.InvalidParameterException;
+import me.itzg.helpers.fabric.FabricLauncherInstaller;
+import me.itzg.helpers.files.IoStreams;
+import me.itzg.helpers.forge.ForgeInstaller;
+import me.itzg.helpers.http.SharedFetchArgs;
+import me.itzg.helpers.json.ObjectMappers;
+import me.itzg.helpers.modrinth.model.*;
+import me.itzg.helpers.quilt.QuiltInstaller;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+import reactor.core.scheduler.Schedulers;
+
+@Slf4j
+public class ModrinthPackInstaller {
+    private final ModrinthApiClient apiClient;
+    private final Path zipFile;
+    private final Path outputDirectory;
+    private final Path resultsFile;
+    private final boolean forceModloaderReinstall;
+    private final SharedFetchArgs sharedFetchArgs;
+
+    public ModrinthPackInstaller(InstallModrinthModpackCommand config, Path zipFile) {
```

Comment:
...for completeness, again, don't pass in the "command object". Pass the parameters specifically needed and pass in the pre-built api client rather than building a new one.